src/extension.ts

```diff
@@ -0,0 +1,140 @@
+                })
+            )
+        }
+    })
+}
+
+/**
+ * Extract Sentry params from document URI necessary to
+ * check if the current document sends log events to Sentry and
+ * use these params build URL to the Sentry issues stream page.
+ * @param textDocument A document URI.
+ * @returns repo and file part of URI.
+ */
+export function getParamsFromUriPath(textDocument: string): Params {
+    // TODO: Support more than just GitHub.
+    const repoPattern = /github\.com\/([^\?\#]+)/gi
+    const filePattern = /#([^\?\#\/]+)\/.*\.?$/gi
+    const repoM = repoPattern.exec(textDocument)
+    const fileM = filePattern.exec(textDocument)
+    return {
+        repo: repoM![1],
+        file: fileM![0],
+    }
+}
+
+/**
+ * Verify if the params from the document URI match with the repo and file formats specified
+ * in the Sentry extension settings. If there is a match we know the document is enabled to send logs
+ * to Sentry and can send back the corresponding Sentry project ID.
+ * @param params params extracted from the document's URI.
+ * @param projects Sentry extension projects configurations.
+ * @return Sentry projectID this document reports to.
+ */
+export function matchSentryProject(params: Params, projects: Settings['sentry.projects']): SentryProject | undefined {
+    // Check if a Sentry project is associated with this document's repo and retrieve the project.
+    const project = projects!.find(p => !!new RegExp(p.patternProperties.repoMatch).exec(params.repo!))
+    if (!project) {
+        return
+    }
+
+    // Check if document matches the file matching pattern specified under the Sentry extension configuration.
+    const fileMatched: boolean = !!project.patternProperties.fileMatches.find(
+        pattern => !!new RegExp(pattern).exec(params.file!)
+    )
+    if (!fileMatched) {
+        return
+    }
+
+    return {
+        projectId: project.projectId,
+        lineMatches: project.patternProperties.lineMatches,
+    }
+}
+
+function decorateEditor(editor: sourcegraph.CodeEditor, sentryProjectId: string, lineMatches: RegExp[]): void {
+    const decorations: sourcegraph.TextDocumentDecoration[] = []
+    for (const [i, line] of editor.document.text!.split('\n').entries()) {
+        let m: RegExpExecArray | null
+        for (let pattern of lineMatches) {
+            pattern = new RegExp(pattern, 'gi')
+            do {
+                m = pattern.exec(line)
+                if (m) {
+                    decorations.push({
+                        range: new sourcegraph.Range(i, 0, i, 0),
+                        isWholeLine: true,
+                        after: {
+                            backgroundColor: '#e03e2f',
+                            color: 'rgba(255, 255, 255, 0.8)',
+                            contentText: ' View logs in Sentry » ',
+                            // Depending on the line matching pattern the query is indexed in position 1 or 2.
```

Comment:
For now this is true, but eventually you want to be able to specify which capture group should be used through configuration.

Reply:
Added that as a TODO.